### DISCUSSION TITLE
JIT selector grouping

### DISCUSSION THREAD

**Discussion by gustavorino** [tailwindlabs/tailwindcss#4192] [Ideas]
I think it would be a great addition to the JIT compiler if we could write expressions like this:

```html
<button class="md:dark:disabled:focus:hover:(bg-gray-400,text-white)">
```

Or simpler:

```html
<button class="sm:(text-white,bg-black,text-2) md:(text-black,bg-transparent,text-4) ">
```

What do you think?

**Comment by NikitaIT** [comment]
To understand how much this is possible, you need to think about things like this.

If we have:
```html
<span class="md:(text-black,bg-transparent,text-4) "/>
```

This will compile to(no special characters for brevity):
```css
.md:(text-black,bg-transparent,text-4) {
   @apply md:text-black md:bg-transparent md:text-4
}
```

**What problems can there be with this?**

1. Will your bundle grow? - YES
2. How much will it grow?  - ...
3. Perhaps you want to do the composition differently.
    - `.text:(red, md:black)` - why not:?
    - `.article-text` - ok, its may be domain driven, the only right way
    
This can be a little confusing, but you shouldn't really leave a lot of tailwind classes in the html. The documentation says a lot about SOC. But keeping abstraction levels is more important. 
- You always sacrifice bundle size for the DSL extension. 
- You should never sacrifice bundle size for syntactic sugar.

**Reply by lucasrabiec** [reply]
@NikitaIT does bundle size also grows when arbitrary variants are used? Like: `[&:is(:hover,:focus,:active)]:text-blue-600`

**Comment by NikitaIT** [comment]
Related https://github.com/tailwindlabs/tailwindcss/discussions/4197
